Packs/AggregatedScripts/Scripts/QuarantineFile/QuarantineFile.py

```diff
@@ -0,0 +1,1145 @@
+
+        It gets the correct handler for the brand, validates arguments, runs pre-checks,
+        and initiates the quarantine action, creating a new pending job if necessary.
+
+        Args:
+            brand (str): The brand to process.
+            endpoint_ids (list): The list of endpoint IDs for this brand.
+        """
+        demisto.debug(f"[Orchestrator] Processing {len(endpoint_ids)} endpoints for brand '{brand}'.")
+        try:
+            handler = handler_factory(brand, self)
+            brand_args = self.args.copy()
+            brand_args[self.ENDPOINT_IDS_ARG] = endpoint_ids
+            handler.validate_args(brand_args)
+            endpoints_to_poll, initial_results = handler.run_pre_checks_and_get_initial_results(brand_args)
+            self.completed_results.extend(initial_results)
+            if endpoints_to_poll:
+                demisto.debug(f"[Orchestrator] {len(endpoints_to_poll)} endpoints for '{brand}' need quarantine action.")
+                initiate_args = self.args.copy()
+                initiate_args[self.ENDPOINT_IDS_ARG] = endpoints_to_poll
+                new_job = handler.initiate_quarantine(initiate_args)
+                self.pending_jobs.append(new_job)
+            return
+        except DemistoException as e:
+            demisto.error(f"Failed to process endpoints for brand '{brand}': {e}")
+            error_msg = QuarantineResult.Messages.FAILED_WITH_REASON.format(reason=e)
+        except Exception as e:
+            demisto.error(f"Failed to process endpoints for brand '{brand}': {e}")
+            error_msg = QuarantineResult.Messages.GENERAL_FAILURE
+
+        for endpoint_id in endpoint_ids:
+            self.completed_results.append(
+                QuarantineResult.create(
+                    endpoint_id=endpoint_id,
+                    status=QuarantineResult.Statuses.FAILED,
+                    message=error_msg,
+                    brand=brand,
+                    script_args=self.args,
+                )
+            )
+
+    def _check_pending_jobs(self):
+        """
+        Handles a polling run: checks the status of all pending jobs.
+
+        For each job, it executes the polling command. If the job is still running,
+        it is kept in the pending list. If it has finished, it is finalized, and
+        the results are collected.
+        """
+        demisto.debug(f"[Orchestrator] Checking status of {len(self.pending_jobs)} pending jobs.")
+        remaining_jobs = []
+        for job in self.pending_jobs:
+            demisto.debug(f"[Orchestrator] Polling job for brand '{job['brand']}'.")
+            demisto.debug(f"[Orchestrator] The Job: {job}")
+            poll_cmd = Command(name=job["poll_command"], args=job["poll_args"], brand=job["brand"])
+            raw_response, verbose_res = poll_cmd.execute()
+            if self.verbose:
+                self.verbose_results.extend(verbose_res)
```

Comment:
how is debug mode file handled? shouldnt it be handled similiarily?
We should also watch out for ExecutionMetrics and other stuff that might come back

Reply:
not sure what the final answer is here, we need to handle other kinds of not relevant entries and handle them in a standard way. lets advise with @YuvHayun